Assets/HoloToolkit/Common/Scripts/Singleton.cs

```diff
@@ -26,20 +27,29 @@ public static T Instance
         {
             get
             {
-                if (!IsInitialized && searchForInstance)
+                if (IsInitialized)
                 {
-                    searchForInstance = false;
-                    T[] objects = FindObjectsOfType<T>();
-                    if (objects.Length == 1)
-                    {
-                        instance = objects[0];
-                        instance.gameObject.GetParentRoot().DontDestroyOnLoad();
-                    }
-                    else if (objects.Length > 1)
-                    {
-                        Debug.LogErrorFormat("Expected exactly 1 {0} but found {1}.", typeof(T).Name, objects.Length);
-                    }
+                    return instance;
                 }
+
+                if (!searchForInstance)
+                {
+                    return null;
+                }
+
+                // likely first time
+                T[] objects = FindObjectsOfType<T>();
+                searchForInstance = false;
+
```

Comment:
We could check to see if we need to instantiate a new instance here by searching for or creating a new `GameObject` to attach this component to.

For example:
```
if(objects.Length==0)
{
    instance = new GameObject(typeof(T).Name).AddComponent<T>();
}
```

Reply:
You are welcome to add new functionality in a new PR, that is out of scope of this PR.